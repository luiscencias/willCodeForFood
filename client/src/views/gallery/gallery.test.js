import React from 'react';
import Gallery from './gallery';
import {shallow} from "enzyme/build";
import { Zoom } from 'react-slideshow-image';


describe('Gallery', () => {

    let gallery;
    beforeEach(() => gallery = shallow(<Gallery />));

    it('renders', () => {

        expect(gallery.exists()).toBe(true);
    });



    it('renders without changing', () => {
        expect(gallery).toMatchSnapshot();
        // console.log(gallery.debug());
    });

    it('should render Zoom component', () => {
        expect(gallery.contains(<Zoom scale={.4}/>));   //scale is required
        // console.log(gallery.debug());
    });

    it('should render an img element', () => {

        expect(gallery.contains(<img/>));
    });



    it('should have js implementation', () => {
        // console.log(gallery.debug());

        expect(gallery.find('.js')).toBeTruthy();
    });



});

