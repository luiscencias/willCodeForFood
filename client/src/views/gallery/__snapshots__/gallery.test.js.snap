// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery renders without changing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Gallery />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="slide-container"
      >
        <Zoom
          arrows={false}
          autoplay={true}
          duration={5000}
          indicators={true}
          infinite={true}
          scale={0.4}
          transitionDuration={500}
        >
          <img
            src="https://instagram.fftw1-1.fna.fbcdn.net/vp/ddbcd4e30ed5d35940d63de642788de3/5D13CFD5/t51.2885-15/e35/52713135_2323907557932378_8977152329679767459_n.jpg?_nc_ht=instagram.fftw1-1.fna.fbcdn.net"
            style={
              Object {
                "height": "70%",
                "marginLeft": "auto",
                "marginRight": "auto",
                "width": "70%",
              }
            }
          />
          <img
            src="https://instagram.fftw1-1.fna.fbcdn.net/vp/3eb89330c0733586b61e955d2fc79b6d/5D0DAED6/t51.2885-15/e35/41739254_2473917379293001_5623219170017712919_n.jpg?_nc_ht=instagram.fftw1-1.fna.fbcdn.net"
            style={
              Object {
                "height": "70%",
                "marginLeft": "auto",
                "marginRight": "auto",
                "width": "70%",
              }
            }
          />
          <img
            src="https://instagram.fftw1-1.fna.fbcdn.net/vp/b0b4577162021c1285d7b2b6eedc2a24/5D23B269/t51.2885-15/e35/52001337_824263094592003_121996381511065480_n.jpg?_nc_ht=instagram.fftw1-1.fna.fbcdn.net"
            style={
              Object {
                "height": "70%",
                "marginLeft": "auto",
                "marginRight": "auto",
                "width": "70%",
              }
            }
          />
        </Zoom>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Zoom
          arrows={false}
          autoplay={true}
          duration={5000}
          indicators={true}
          infinite={true}
          scale={0.4}
          transitionDuration={500}
        >
          <img
            src="https://instagram.fftw1-1.fna.fbcdn.net/vp/ddbcd4e30ed5d35940d63de642788de3/5D13CFD5/t51.2885-15/e35/52713135_2323907557932378_8977152329679767459_n.jpg?_nc_ht=instagram.fftw1-1.fna.fbcdn.net"
            style={
              Object {
                "height": "70%",
                "marginLeft": "auto",
                "marginRight": "auto",
                "width": "70%",
              }
            }
          />
          <img
            src="https://instagram.fftw1-1.fna.fbcdn.net/vp/3eb89330c0733586b61e955d2fc79b6d/5D0DAED6/t51.2885-15/e35/41739254_2473917379293001_5623219170017712919_n.jpg?_nc_ht=instagram.fftw1-1.fna.fbcdn.net"
            style={
              Object {
                "height": "70%",
                "marginLeft": "auto",
                "marginRight": "auto",
                "width": "70%",
              }
            }
          />
          <img
            src="https://instagram.fftw1-1.fna.fbcdn.net/vp/b0b4577162021c1285d7b2b6eedc2a24/5D23B269/t51.2885-15/e35/52001337_824263094592003_121996381511065480_n.jpg?_nc_ht=instagram.fftw1-1.fna.fbcdn.net"
            style={
              Object {
                "height": "70%",
                "marginLeft": "auto",
                "marginRight": "auto",
                "width": "70%",
              }
            }
          />
        </Zoom>,
        "className": "slide-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "arrows": false,
          "autoplay": true,
          "children": Array [
            <img
              src="https://instagram.fftw1-1.fna.fbcdn.net/vp/ddbcd4e30ed5d35940d63de642788de3/5D13CFD5/t51.2885-15/e35/52713135_2323907557932378_8977152329679767459_n.jpg?_nc_ht=instagram.fftw1-1.fna.fbcdn.net"
              style={
                Object {
                  "height": "70%",
                  "marginLeft": "auto",
                  "marginRight": "auto",
                  "width": "70%",
                }
              }
            />,
            <img
              src="https://instagram.fftw1-1.fna.fbcdn.net/vp/3eb89330c0733586b61e955d2fc79b6d/5D0DAED6/t51.2885-15/e35/41739254_2473917379293001_5623219170017712919_n.jpg?_nc_ht=instagram.fftw1-1.fna.fbcdn.net"
              style={
                Object {
                  "height": "70%",
                  "marginLeft": "auto",
                  "marginRight": "auto",
                  "width": "70%",
                }
              }
            />,
            <img
              src="https://instagram.fftw1-1.fna.fbcdn.net/vp/b0b4577162021c1285d7b2b6eedc2a24/5D23B269/t51.2885-15/e35/52001337_824263094592003_121996381511065480_n.jpg?_nc_ht=instagram.fftw1-1.fna.fbcdn.net"
              style={
                Object {
                  "height": "70%",
                  "marginLeft": "auto",
                  "marginRight": "auto",
                  "width": "70%",
                }
              }
            />,
          ],
          "duration": 5000,
          "indicators": true,
          "infinite": true,
          "scale": 0.4,
          "transitionDuration": 500,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "src": "https://instagram.fftw1-1.fna.fbcdn.net/vp/ddbcd4e30ed5d35940d63de642788de3/5D13CFD5/t51.2885-15/e35/52713135_2323907557932378_8977152329679767459_n.jpg?_nc_ht=instagram.fftw1-1.fna.fbcdn.net",
              "style": Object {
                "height": "70%",
                "marginLeft": "auto",
                "marginRight": "auto",
                "width": "70%",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "src": "https://instagram.fftw1-1.fna.fbcdn.net/vp/3eb89330c0733586b61e955d2fc79b6d/5D0DAED6/t51.2885-15/e35/41739254_2473917379293001_5623219170017712919_n.jpg?_nc_ht=instagram.fftw1-1.fna.fbcdn.net",
              "style": Object {
                "height": "70%",
                "marginLeft": "auto",
                "marginRight": "auto",
                "width": "70%",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "host",
            "props": Object {
              "src": "https://instagram.fftw1-1.fna.fbcdn.net/vp/b0b4577162021c1285d7b2b6eedc2a24/5D23B269/t51.2885-15/e35/52001337_824263094592003_121996381511065480_n.jpg?_nc_ht=instagram.fftw1-1.fna.fbcdn.net",
              "style": Object {
                "height": "70%",
                "marginLeft": "auto",
                "marginRight": "auto",
                "width": "70%",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="slide-container"
        >
          <Zoom
            arrows={false}
            autoplay={true}
            duration={5000}
            indicators={true}
            infinite={true}
            scale={0.4}
            transitionDuration={500}
          >
            <img
              src="https://instagram.fftw1-1.fna.fbcdn.net/vp/ddbcd4e30ed5d35940d63de642788de3/5D13CFD5/t51.2885-15/e35/52713135_2323907557932378_8977152329679767459_n.jpg?_nc_ht=instagram.fftw1-1.fna.fbcdn.net"
              style={
                Object {
                  "height": "70%",
                  "marginLeft": "auto",
                  "marginRight": "auto",
                  "width": "70%",
                }
              }
            />
            <img
              src="https://instagram.fftw1-1.fna.fbcdn.net/vp/3eb89330c0733586b61e955d2fc79b6d/5D0DAED6/t51.2885-15/e35/41739254_2473917379293001_5623219170017712919_n.jpg?_nc_ht=instagram.fftw1-1.fna.fbcdn.net"
              style={
                Object {
                  "height": "70%",
                  "marginLeft": "auto",
                  "marginRight": "auto",
                  "width": "70%",
                }
              }
            />
            <img
              src="https://instagram.fftw1-1.fna.fbcdn.net/vp/b0b4577162021c1285d7b2b6eedc2a24/5D23B269/t51.2885-15/e35/52001337_824263094592003_121996381511065480_n.jpg?_nc_ht=instagram.fftw1-1.fna.fbcdn.net"
              style={
                Object {
                  "height": "70%",
                  "marginLeft": "auto",
                  "marginRight": "auto",
                  "width": "70%",
                }
              }
            />
          </Zoom>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Zoom
            arrows={false}
            autoplay={true}
            duration={5000}
            indicators={true}
            infinite={true}
            scale={0.4}
            transitionDuration={500}
          >
            <img
              src="https://instagram.fftw1-1.fna.fbcdn.net/vp/ddbcd4e30ed5d35940d63de642788de3/5D13CFD5/t51.2885-15/e35/52713135_2323907557932378_8977152329679767459_n.jpg?_nc_ht=instagram.fftw1-1.fna.fbcdn.net"
              style={
                Object {
                  "height": "70%",
                  "marginLeft": "auto",
                  "marginRight": "auto",
                  "width": "70%",
                }
              }
            />
            <img
              src="https://instagram.fftw1-1.fna.fbcdn.net/vp/3eb89330c0733586b61e955d2fc79b6d/5D0DAED6/t51.2885-15/e35/41739254_2473917379293001_5623219170017712919_n.jpg?_nc_ht=instagram.fftw1-1.fna.fbcdn.net"
              style={
                Object {
                  "height": "70%",
                  "marginLeft": "auto",
                  "marginRight": "auto",
                  "width": "70%",
                }
              }
            />
            <img
              src="https://instagram.fftw1-1.fna.fbcdn.net/vp/b0b4577162021c1285d7b2b6eedc2a24/5D23B269/t51.2885-15/e35/52001337_824263094592003_121996381511065480_n.jpg?_nc_ht=instagram.fftw1-1.fna.fbcdn.net"
              style={
                Object {
                  "height": "70%",
                  "marginLeft": "auto",
                  "marginRight": "auto",
                  "width": "70%",
                }
              }
            />
          </Zoom>,
          "className": "slide-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "arrows": false,
            "autoplay": true,
            "children": Array [
              <img
                src="https://instagram.fftw1-1.fna.fbcdn.net/vp/ddbcd4e30ed5d35940d63de642788de3/5D13CFD5/t51.2885-15/e35/52713135_2323907557932378_8977152329679767459_n.jpg?_nc_ht=instagram.fftw1-1.fna.fbcdn.net"
                style={
                  Object {
                    "height": "70%",
                    "marginLeft": "auto",
                    "marginRight": "auto",
                    "width": "70%",
                  }
                }
              />,
              <img
                src="https://instagram.fftw1-1.fna.fbcdn.net/vp/3eb89330c0733586b61e955d2fc79b6d/5D0DAED6/t51.2885-15/e35/41739254_2473917379293001_5623219170017712919_n.jpg?_nc_ht=instagram.fftw1-1.fna.fbcdn.net"
                style={
                  Object {
                    "height": "70%",
                    "marginLeft": "auto",
                    "marginRight": "auto",
                    "width": "70%",
                  }
                }
              />,
              <img
                src="https://instagram.fftw1-1.fna.fbcdn.net/vp/b0b4577162021c1285d7b2b6eedc2a24/5D23B269/t51.2885-15/e35/52001337_824263094592003_121996381511065480_n.jpg?_nc_ht=instagram.fftw1-1.fna.fbcdn.net"
                style={
                  Object {
                    "height": "70%",
                    "marginLeft": "auto",
                    "marginRight": "auto",
                    "width": "70%",
                  }
                }
              />,
            ],
            "duration": 5000,
            "indicators": true,
            "infinite": true,
            "scale": 0.4,
            "transitionDuration": 500,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "src": "https://instagram.fftw1-1.fna.fbcdn.net/vp/ddbcd4e30ed5d35940d63de642788de3/5D13CFD5/t51.2885-15/e35/52713135_2323907557932378_8977152329679767459_n.jpg?_nc_ht=instagram.fftw1-1.fna.fbcdn.net",
                "style": Object {
                  "height": "70%",
                  "marginLeft": "auto",
                  "marginRight": "auto",
                  "width": "70%",
                },
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "src": "https://instagram.fftw1-1.fna.fbcdn.net/vp/3eb89330c0733586b61e955d2fc79b6d/5D0DAED6/t51.2885-15/e35/41739254_2473917379293001_5623219170017712919_n.jpg?_nc_ht=instagram.fftw1-1.fna.fbcdn.net",
                "style": Object {
                  "height": "70%",
                  "marginLeft": "auto",
                  "marginRight": "auto",
                  "width": "70%",
                },
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "host",
              "props": Object {
                "src": "https://instagram.fftw1-1.fna.fbcdn.net/vp/b0b4577162021c1285d7b2b6eedc2a24/5D23B269/t51.2885-15/e35/52001337_824263094592003_121996381511065480_n.jpg?_nc_ht=instagram.fftw1-1.fna.fbcdn.net",
                "style": Object {
                  "height": "70%",
                  "marginLeft": "auto",
                  "marginRight": "auto",
                  "width": "70%",
                },
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
          ],
          "type": [Function],
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
